libcontainer/cgroups/systemd/v1.go

```diff
@@ -326,6 +332,57 @@ func (m *legacyManager) GetStats() (*cgroups.Stats, error) {
 	return stats, nil
 }
 
+// freezeBeforeSet answers whether there is a need to freeze the cgroup before
+// applying its systemd unit properties, and thaw after, while avoiding
+// unnecessary freezer state changes.
+//
+// The reason why we have to freeze is that systemd's application of device
+// rules is done disruptively, resulting in spurious errors to common devices
+// (unlike our fs driver, they will happily write deny-all rules to running
+// containers). So we have to freeze the container to avoid the container get
+// an occasional "permission denied" error.
+func (m *legacyManager) freezeBeforeSet(unitName string, r *configs.Resources) (needsFreeze, needsThaw bool, Err error) {
+	// Special case for SkipDevices.
```

Comment:
I would mention in this comment that this is also a special case where the systemd cgroup has been configured with an allow-all device policy (and that this situation is pretty much Kubernetes-specific), to make sure we don't actually expand this behaviour in the future.

Reply:
amended the comment.